### PR TITLE
remove redundant code path

### PR DESCRIPTION
**Pull request by bekzod**

### CODE REVIEW

File: lib/rsvp/enumerator.js [hunk 1]
```diff
@@ -63,10 +61,6 @@ export default class Enumerator {
         this._settledAt(entry._state, i, entry._result, firstPass);
       } else if (typeof then !== 'function') {
         this._settledAt(FULFILLED, i, entry, firstPass);
-      } else if (this._isUsingOwnPromise) {
-        let promise = new c(noop);
-        handleMaybeThenable(promise, entry, then);
-        this._willSettleAt(promise, i, firstPass);
       } else {
```
Comment (bekzod) [inline lib/rsvp/enumerator.js]: `new c(resolve => resolve(entry)` would have same code path internally calling `handleMaybeThenable(promise, entry, then);`

File: lib/rsvp/enumerator.js [hunk 2]
```diff
@@ -18,9 +17,8 @@ import ownResolve from './promise/resolve';
 export default class Enumerator {
   constructor(Constructor, input, abortOnReject, label) {
     this._instanceConstructor = Constructor;
-    this.promise = new Constructor(noop, label);
+    this.promise = new OwnPromise(noop, label);
```
Comment (bekzod) [inline lib/rsvp/enumerator.js]: this should be always `OwnPromise` else `subscribe/publish` system won't work

File: lib/rsvp/enumerator.js [hunk 3]
```diff
@@ -63,10 +61,6 @@ export default class Enumerator {
         this._settledAt(entry._state, i, entry._result, firstPass);
       } else if (typeof then !== 'function') {
         this._settledAt(FULFILLED, i, entry, firstPass);
-      } else if (this._isUsingOwnPromise) {
-        let promise = new c(noop);
-        handleMaybeThenable(promise, entry, then);
-        this._willSettleAt(promise, i, firstPass);
```
Comment (bekzod) [inline lib/rsvp/enumerator.js]: new c(resolve => resolve(entry) would have same code path internally calling handleMaybeThenable(promise, entry, then);

File: lib/rsvp/enumerator.js (hunk 2 above)
Comment (stefanpenner) [reply lib/rsvp/enumerator.js]: This doesn't seem right.

The following must remain true `Subclass.all([1]).then() instanceof Subclass`

**Comment by wagenet** [comment]
Closing due to staleness.

**Pull request closed without merging**